domain/src/Aggregates/TaxYear/Services/TaxYearNormaliser/TaxYearNormaliser.php

```diff
@@ -4,10 +4,22 @@
 
 namespace Domain\Aggregates\TaxYear\Services\TaxYearNormaliser;
 
+use Brick\DateTime\LocalDate;
+
 final class TaxYearNormaliser
 {
-    public static function fromYear(int $year): string
+    private const APRIL = 4;
+
+    public static function fromDate(LocalDate $date): string
     {
+        $year = $date->getYear();
+        $month = $date->getMonth();
+        $day = $date->getDay();
+
+        if ($month < self::APRIL || ($month === self::APRIL && $day < 6)) {
+            --$year;
+        }
+
         return sprintf('%s-%s', $year, $year + 1);
```

Comment:
ℹ️ This logic should of course determine the tax year also based on the month and day (before or after April 5, essentially)